src/go/rpk/pkg/cli/cluster/config/list.go

```diff
@@ -0,0 +1,215 @@
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/publicapi"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newListCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var filter string
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List cluster configuration properties",
+		Long: `List cluster configuration properties.
+
+This command lists all available cluster configuration properties. Use the 'get'
+command to retrieve specific property values, or 'edit' for interactive editing.
+
+Use the --filter flag with a regular expression to filter configuration keys.`,
+		Example: `
+List all cluster configuration properties:
+  rpk cluster config list
+
+List configuration properties matching a filter:
+  rpk cluster config list --filter="kafka.*"
+
+List configuration properties in JSON format:
+  rpk cluster config list --format=json
+
+List configuration properties in YAML format:
+  rpk cluster config list --format=yaml`,
+		Args: cobra.NoArgs,
+		Run: func(cmd *cobra.Command, _ []string) {
+			f := p.Formatter
+			if h, ok := f.Help([]ConfigProperty{}); ok {
+				out.Exit(h)
+			}
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+
+			p := cfg.VirtualProfile()
+
+			var configList []ConfigProperty
+
+			if p.FromCloud {
+				if p.CloudCluster.IsServerless() {
+					out.Die("rpk cluster config list is not supported for serverless clusters")
+				}
+				configList, err = listCloudConfig(cmd.Context(), cfg, p)
+				out.MaybeDie(err, "rpk unable to list cloud config: %v", err)
+			} else {
+				client, err := adminapi.NewClient(cmd.Context(), fs, p)
+				out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+				currentConfig, err := client.Config(cmd.Context(), true)
+				out.MaybeDie(err, "unable to query current config: %v", err)
+
+				for key, value := range currentConfig {
+					configList = append(configList, ConfigProperty{
+						Key:   key,
+						Value: formatValueForDisplay(value),
+					})
+				}
+			}
+
+			// Apply regex filter if provided
+			if filter != "" {
+				filteredList, err := filterConfigList(configList, filter)
+				out.MaybeDie(err, "unable to apply filter: %v", err)
+				configList = filteredList
+			}
+
+			// Sort by key for consistent output
+			sort.Slice(configList, func(i, j int) bool {
+				return configList[i].Key < configList[j].Key
+			})
+
+			if isText, _, s, err := f.Format(configList); !isText {
+				out.MaybeDie(err, "unable to print in the required format %q: %v", f.Kind, err)
+				out.Exit(s)
+				return
+			}
+			t := out.NewTable("Property", "Value")
+			defer t.Flush()
+			for _, prop := range configList {
+				t.PrintStructFields(prop)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&filter, "filter", "", "Filter configuration keys using regular expression")
+	p.InstallFormatFlag(cmd)
+	return cmd
+}
+
+type ConfigProperty struct {
+	Key   string `json:"key" yaml:"key"`
+	Value any    `json:"value" yaml:"value"`
+}
```

Comment:
I don't think we should expose an array of key-value objects, instead I think we should expose a map:

instead of:
```
[
  {"key": "log_segment_size", "value": "20971520"},
  {"key": "retention_local_target_ms_default", "value": "21600000"},
  {"key":...}
]
```

we should instead do
```
{
  "log_segment_size": "20971520",
  "retention_local_target_ms_default": "21600000"
   ...
}
```

It is faster to access, doesn't require iteration when looking for a key, and is cleaner (as it is not nested). It's also easier to use with other tools (as jq)

Was there a reason to use this struct?